### PR TITLE
Clean up self-set node labels

### PR DESCRIPTION
**Pull request by liggitt**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Starts to clean up self-set kubelet labels under k8s.io prefixes:
* removes fluentd node selector to complete migration to daemonset
* starts the process of switching the `beta.kubernetes.io/metadata-proxy-ready` label to a `cloud.google.com/metadata-proxy-ready` label by adding the label to new nodes and labeling existing nodes on upgrade
* starts the process of switching the `beta.kubernetes.io/kube-proxy-ds-ready` label to a `node.kubernetes.io/kube-proxy-ds-ready` label by adding the label to new nodes and labeling existing nodes on upgrade
* starts the process of switching the `beta.kubernetes.io/masq-agent-ds-ready` label to a `node.kubernetes.io/masq-agent-ds-ready` label by adding the label to new nodes and labeling existing nodes on upgrade

This is related to https://github.com/kubernetes/enhancements/blob/master/keps/sig-auth/0000-20170814-bounding-self-labeling-kubelets.md#implementation-timeline

**Does this PR introduce a user-facing change?**:
```release-note
* the fluentd addon daemonset will now target all nodes.
* setting `ENABLE_METADATA_CONCEALMENT=true` in kube-up will now set a `cloud.google.com/metadata-proxy-ready=true` label on new nodes. In v1.16, the metadata proxy add-on will switch to using that label as a node selector.
* setting `KUBE_PROXY_DAEMONSET=true` in kube-up will now set a `node.kubernetes.io/kube-proxy-ds-ready=true` label on new nodes. In v1.16, the kube-proxy daemonset add-on will switch to using that label as a node selector.
* In 1.16, the masq-agent daemonset add-on will switch to using `node.kubernetes.io/masq-agent-ds-ready` as a node selector.
```

/cc @piosz
for fluentd cleanup

/cc @dekkagaijin 
for metadata proxy label

/sig cluster-lifecycle
/sig gcp

### CODE REVIEW

**Comment by liggitt** [comment]
/cc @MrHohn 
for kube-proxy daemonset labeling

File: cluster/gce/gci/configure-helper.sh [hunk 1]
```diff
@@ -1927,6 +1927,35 @@ function setup-etcd-encryption {
   fi
 }
 
+# Updates node labels used by addons.
+function update-legacy-addon-node-labels() {
```
Comment (MrHohn) [inline cluster/gce/gci/configure-helper.sh]: nit: probably add a log line for this step?

Comment (MrHohn) [reply cluster/gce/gci/configure-helper.sh]: To make sure I understand it correctly --- this step is needed for the upgrade case, in which kubelet won't be able to update existing node to have the new label even if it is given via `--node-labels`?

**Review by MrHohn — APPROVED** [review]
LGTM for kube-proxy and ip-masq-agent

File: cluster/gce/gci/configure-helper.sh (hunk 1 above)
Comment (liggitt) [reply cluster/gce/gci/configure-helper.sh]: > this step is needed for the upgrade case, in which kubelet won't be able to update existing node to have the new label even if it is given via `--node-labels`?

exactly

**Comment by liggitt** [comment]
/hold

would like @dekkagaijin to weigh in on the metadata proxy change

**Comment by soggiest** [comment]
@dekkagaijin please review per @liggitt's comment above. Code freeze for 1.14 is in 3 days, this will need to be merged within the week.

**Comment by liggitt** [comment]
also want @thockin's thoughts on the `cloud.google.com/metadata-proxy-ready` label. That was intended to match things like `cloud.google.com/gke-accelerator`, but I'm not sure if that matches current thinking

/assign @thockin

**Comment by thockin** [comment]
I have precious little knowledge of the metadata proxy stuff.  All of these make me want to hurt myself.  Thanks for cleaning up.  Do we have a plan as to when we can actually get rid of them?  I know proxy daemonset work stalled..

**Comment by thockin** [comment]
@mikedanese for metdata viz, too

**Comment by thockin** [comment]
/lgtm

on Jake's behalf

**Comment by liggitt** [comment]
/hold cancel

> Do we have a plan as to when we can actually get rid of them? I know proxy daemonset work stalled..

we will need to keep selectors long term for the things that legitimately may not want to run on every node (masq-agent and metadata-proxy)

I don't know if there were blockers to kube-proxy progressing to run as a DS or if that work just got preempted

**Pull request merged**